cuqi/distribution/_implicit_regularized_gaussian.py

```diff
@@ -0,0 +1,157 @@
+from cuqi.utilities import get_non_default_args
+from cuqi.distribution import Distribution, Gaussian, GMRF
+from cuqi.solver import ProjectNonnegative, ProjectBox, ProximalL1
+
+import numpy as np
+
+
+class ImplicitRegularizedGaussian(Distribution):
+    def __init__(self, gaussian, proximal = None, projector = None, constraint = None, regularization = None, **kwargs):

```

Comment:
Could you add a docstring that explains this distribution and its parameters? Look e.g. at Gaussian or GMRF docstring.